roles/jd-client/src/lib/downstream.rs

```diff
@@ -376,12 +376,12 @@ impl DownstreamMiningNode {
         let to_send = to_send.into_values();
         for message in to_send {
             let message = if let Mining::NewExtendedMiningJob(job) = message {
-                let jd = self_mutex.safe_lock(|s| s.jd.clone()).unwrap().unwrap();
-                jd.safe_lock(|jd| jd.coinbase_tx_prefix = job.coinbase_tx_prefix.clone())
-                    .unwrap();
-                jd.safe_lock(|jd| jd.coinbase_tx_suffix = job.coinbase_tx_suffix.clone())
-                    .unwrap();
-
+                if let Some(jd) = self_mutex.safe_lock(|s| s.jd.clone()).unwrap() {
+                    jd.safe_lock(|jd| jd.coinbase_tx_prefix = job.coinbase_tx_prefix.clone())
+                        .unwrap();
+                    jd.safe_lock(|jd| jd.coinbase_tx_suffix = job.coinbase_tx_suffix.clone())
+                        .unwrap();
```

Comment:
```suggestion
                     jd.safe_lock(|jd| {
                        jd.coinbase_tx_prefix = job.coinbase_tx_prefix.clone();
                        jd.coinbase_tx_suffix = job.coinbase_tx_suffix.clone();
                    }).unwrap();
```

Reply:
I don't commit this suggestion because of the policies on how nice commit history must be, if you need credits for this mod I can add it in the commit message. Let me know.

Reply:
No credit needed. I see you included the suggestion in the latest push, and it works perfectly.